rebasehelper/tags.py

```diff
@@ -86,12 +86,17 @@ def sanitize(name):
             return name.capitalize()
         result = []
         tag_re = re.compile(r'^(?P<prefix>(?P<name>\w+)\s*:\s*)(?P<value>.+)$')
-        for section, _ in raw_content.sections:
+        parsed_idx = 0
+        for section, section_content in raw_content.sections:
             section = section.lower()
             if not section.startswith('%package'):
                 continue
-            parsed = parsed_content.section(section)
-            for index, line in enumerate(raw_content.section(section)):
+            if parsed_idx >= len(parsed_content.sections) or section != parsed_content.sections[parsed_idx][0].lower():
+                parsed: List[str] = []
+            else:
+                parsed = parsed_content.sections[parsed_idx][1]
+                parsed_idx += 1
```

Comment:
I don't like this at all. First, it's not obvious what it actually does, second, it's not very robust. I think you should create a mapping between raw and parsed sections in advance, and do lookups in it.